Converted to draft while fixing build error

```
Error: src/kafka-container-7.test.ts(94,9): error TS2322: Type '{ ca: NonSharedBuffer[]; }' is not assignable to type 'boolean | undefined'.
Error: src/kafka-container-7.test.ts(131,9): error TS2322: Type '{ ca: NonSharedBuffer[]; }' is not assignable to type 'boolean | undefined'.
Error: src/kafka-container-latest.test.ts(65,7): error TS2322: Type '{ ca: NonSharedBuffer[]; }' is not assignable to type 'boolean | undefined'.
```

--- a/packages/modules/kafka/package.json
+++ b/packages/modules/kafka/package.json
@@ -29,7 +29,7 @@
     "build": "tsc --project tsconfig.build.json"
   },
   "devDependencies": {
-    "kafkajs": "^2.2.4"
+    "@confluentinc/kafka-javascript": "^1.7.0"
   },
   "dependencies": {
     "compare-versions": "^6.1.1",

--- a/packages/modules/kafka/src/kafka-container-7.test.ts
+++ b/packages/modules/kafka/src/kafka-container-7.test.ts
@@ -84,17 +84,20 @@ describe("KafkaContainer", { timeout: 240_000 }, () => {
         })
         .start();
 
-      await assertMessageProducedAndConsumed(container, {
-        brokers: [`${container.getHost()}:${container.getMappedPort(9096)}`],
-        sasl: {
-          username: "app-user",
-          password: "userPassword",
-          mechanism: "scram-sha-512",
+      await assertMessageProducedAndConsumed(
+        container,
+        {
+          brokers: [`${container.getHost()}:${container.getMappedPort(9096)}`],
+          ssl: true,
         },
-        ssl: {
-          ca: [fs.readFileSync(path.resolve(certificatesDir, "kafka.client.truststore.pem"))],
-        },
-      });
+        {
+          "sasl.mechanism": "SCRAM-SHA-512",
+          "sasl.username": "app-user",
+          "sasl.password": "userPassword",
+          "security.protocol": "sasl_ssl",
+          "ssl.ca.location": path.resolve(certificatesDir, "kafka.client.truststore.pem"),
+        }
+      );
       // }
     });
 
@@ -121,17 +124,20 @@ describe("KafkaContainer", { timeout: 240_000 }, () => {
         })
         .start();
 
-      await assertMessageProducedAndConsumed(container, {
-        brokers: [`${container.getHost()}:${container.getMappedPort(9096)}`],
-        sasl: {
-          username: "app-user",
-          password: "userPassword",
-          mechanism: "scram-sha-512",
+      await assertMessageProducedAndConsumed(
+        container,
+        {
+          brokers: [`${container.getHost()}:${container.getMappedPort(9096)}`],
+          ssl: true,
         },
-        ssl: {
-          ca: [fs.readFileSync(path.resolve(certificatesDir, "kafka.client.truststore.pem"))],
-        },
-      });
+        {
+          "sasl.mechanism": "SCRAM-SHA-512",
+          "sasl.username": "app-user",
+          "sasl.password": "userPassword",
+          "security.protocol": "sasl_ssl",
+          "ssl.ca.location": path.resolve(certificatesDir, "kafka.client.truststore.pem"),
+        }
+      );
     });
 
     it(`should connect within Docker network`, async () => {

--- a/packages/modules/kafka/src/kafka-container-latest.test.ts
+++ b/packages/modules/kafka/src/kafka-container-latest.test.ts
@@ -55,17 +55,20 @@ describe("KafkaContainer", { timeout: 240_000 }, () => {
 
     await using container = await new KafkaContainer(IMAGE).withSaslSslListener(saslConfig).start();
 
-    await assertMessageProducedAndConsumed(container, {
-      brokers: [`${container.getHost()}:${container.getMappedPort(9096)}`],
-      sasl: {
-        username: "app-user",
-        password: "userPassword",
-        mechanism: "scram-sha-512",
+    await assertMessageProducedAndConsumed(
+      container,
+      {
+        brokers: [`${container.getHost()}:${container.getMappedPort(9096)}`],
+        ssl: true,
       },
-      ssl: {
-        ca: [fs.readFileSync(path.resolve(certificatesDir, "kafka.client.truststore.pem"))],
-      },
-    });
+      {
+        "sasl.mechanism": "SCRAM-SHA-512",
+        "sasl.username": "app-user",
+        "sasl.password": "userPassword",
+        "security.protocol": "sasl_ssl",
+        "ssl.ca.location": path.resolve(certificatesDir, "kafka.client.truststore.pem"),
+      }
+    );
     // }
   });
 

--- a/packages/modules/kafka/src/test-helper.ts
+++ b/packages/modules/kafka/src/test-helper.ts
@@ -1,21 +1,29 @@
-import { Kafka, KafkaConfig, logLevel } from "kafkajs";
+import { GlobalConfig, KafkaJS } from "@confluentinc/kafka-javascript";
 import { StartedKafkaContainer } from "./kafka-container";
 
 // kafkaTestHelper {
 export async function assertMessageProducedAndConsumed(
   container: StartedKafkaContainer,
-  additionalConfig: Partial<KafkaConfig> = {}
+  additionalKafkaConfig: Partial<KafkaJS.KafkaConfig> = {},
+  additionalGlobalConfig: Partial<GlobalConfig> = {}
 ) {
   const brokers = [`${container.getHost()}:${container.getMappedPort(9093)}`];
-  const kafka = new Kafka({ logLevel: logLevel.NOTHING, brokers: brokers, ...additionalConfig });
+  const kafka = new KafkaJS.Kafka({
+    kafkaJS: {
+      logLevel: KafkaJS.logLevel.ERROR,
+      brokers,
+      ...additionalKafkaConfig,
+    },
+    ...additionalGlobalConfig,
+  });
 
   const producer = kafka.producer();
   await producer.connect();
-  const consumer = kafka.consumer({ groupId: "test-group" });
+  const consumer = kafka.consumer({ kafkaJS: { groupId: "test-group", fromBeginning: true } });
   await consumer.connect();
 
   await producer.send({ topic: "test-topic", messages: [{ value: "test message" }] });
-  await consumer.subscribe({ topic: "test-topic", fromBeginning: true });
+  await consumer.subscribe({ topic: "test-topic" });
 
   const consumedMessage = await new Promise((resolve) =>
     consumer.run({

--- a/packages/modules/redpanda/package.json
+++ b/packages/modules/redpanda/package.json
@@ -33,6 +33,6 @@
     "testcontainers": "^11.9.0"
   },
   "devDependencies": {
-    "kafkajs": "^2.2.4"
+    "@confluentinc/kafka-javascript": "^1.7.0"
   }
 }

--- a/packages/modules/redpanda/src/test-helper.ts
+++ b/packages/modules/redpanda/src/test-helper.ts
@@ -1,17 +1,22 @@
-import { Kafka, logLevel } from "kafkajs";
+import { KafkaJS } from "@confluentinc/kafka-javascript";
 import { StartedRedpandaContainer } from "./redpanda-container";
 
 // redpandaTestHelper {
 export async function assertMessageProducedAndConsumed(container: StartedRedpandaContainer) {
-  const kafka = new Kafka({ logLevel: logLevel.NOTHING, brokers: [container.getBootstrapServers()] });
+  const kafka = new KafkaJS.Kafka({
+    kafkaJS: {
+      logLevel: KafkaJS.logLevel.NOTHING,
+      brokers: [container.getBootstrapServers()],
+    },
+  });
 
   const producer = kafka.producer();
   await producer.connect();
-  const consumer = kafka.consumer({ groupId: "test-group" });
+  const consumer = kafka.consumer({ kafkaJS: { groupId: "test-group", fromBeginning: true } });
   await consumer.connect();
 
   await producer.send({ topic: "test-topic", messages: [{ value: "test message" }] });
-  await consumer.subscribe({ topic: "test-topic", fromBeginning: true });
+  await consumer.subscribe({ topic: "test-topic" });
 
   const consumedMessage = await new Promise((resolve) =>
     consumer.run({

